x-pack/legacy/plugins/siem/public/pages/case/components/all_cases/index.tsx

```diff
@@ -77,7 +101,17 @@ export const AllCases = React.memo(() => {
     [filterOptions, setFilters]
   );
 
-  const memoizedGetCasesColumns = useMemo(() => getCasesColumns(), []);
+  const updateTheState = useCallback(
+    ({ caseId, version }: Case, updateValue: 'open' | 'closed') => {
+      dispatchUpdateCaseProperty({ updateKey: 'state', updateValue, caseId, version });
+    },
+    []
+  );
+  const actions = useMemo(() => getActions(filterOptions.state, updateTheState), [
```

Comment:
@stephmilovic and I talked and agreed that we can make this more generic by using the dispatch update directly into the `getActions` like 
```
  const actions = useMemo(
    () =>
      getActions({ caseStatus: filterOptions.state, dispatchUpdate: dispatchUpdateCaseProperty }),
    [filterOptions.state, dispatchUpdateCaseProperty]
  );
```
but of course it means to refactor `getActions` like that
```
interface GetActions {
  caseStatus: string;
  dispatchUpdate: Dispatch<UpdateCase>;
}

export const getActions = ({
  caseStatus,
  dispatchUpdate,
}: GetActions): Array<DefaultItemIconButtonAction<Case>> => [
  {
    description: i18n.DELETE,
    icon: 'trash',
    name: i18n.DELETE,
    // eslint-disable-next-line no-console
    onClick: ({ caseId }: Case) => console.log('TO DO Delete case', caseId),
    type: 'icon',
    'data-test-subj': 'action-delete',
  },
  caseStatus === 'open'
    ? {
        description: i18n.CLOSE_CASE,
        icon: 'magnet',
        name: i18n.CLOSE_CASE,
        onClick: (theCase: Case) =>
          dispatchUpdate({
            updateKey: 'state',
            updateValue: 'closed',
            caseId: theCase.caseId,
            version: theCase.version,
          }),
        type: 'icon',
        'data-test-subj': 'action-close',
      }
    : {
        description: i18n.REOPEN_CASE,
        icon: 'magnet',
        name: i18n.REOPEN_CASE,
        onClick: (theCase: Case) =>
          dispatchUpdate({
            updateKey: 'state',
            updateValue: 'open',
            caseId: theCase.caseId,
            version: theCase.version,
          }),
        type: 'icon',
        'data-test-subj': 'action-open',
      },
  {
    description: i18n.DUPLICATE_CASE,
    icon: 'copy',
    name: i18n.DUPLICATE_CASE,
    // eslint-disable-next-line no-console
    onClick: ({ caseId }: Case) => console.log('TO DO Duplicate case', caseId),
    type: 'icon',
    'data-test-subj': 'action-to-do',
  },
];
```